src/test/game/BoardTest.java

```diff
@@ -2,148 +2,223 @@
 
 import main.game.Board;
 import main.game.Line;
+import main.game.Mark;
 import org.junit.Test;
 
+import java.util.EmptyStackException;
 import java.util.List;
 
 import static java.util.Arrays.asList;
+import static main.game.Mark.*;
+import static org.hamcrest.CoreMatchers.is;
 import static org.junit.Assert.*;
 
 public class BoardTest {
 
-    private final String PLAYER_ONE = "X";
-    private final List<String> emptyBoard = asList("1", "2", "3", "4", "5", "6", "7", "8", "9");
-
     @Test
     public void canMakeMoveOnABoard() {
-        Board board = new Board(emptyBoard);
-        List grid = board.updateMove("1", PLAYER_ONE);
+        Board board = new Board(3);
+        List<Mark> grid = board.updateMove(1, X);
 
-        assertEquals(PLAYER_ONE, grid.get(0));
+        assertEquals(X, grid.get(0));
     }
 
     @Test
     public void canReturnRows() {
-        Board board = new Board(emptyBoard);
+        Board board = new Board(3, asList(X, X, X, O, O, O, X, O, X));
+
         List<Line> rows = board.rowLines();
-        assertEquals(createNewLine("1", "2", "3"), rows.get(0));
-        assertEquals(createNewLine("4", "5", "6"), rows.get(1));
-        assertEquals(createNewLine("7", "8", "9"), rows.get(2));
+
+        assertEquals(createNewLine(X, X, X), rows.get(0));
+        assertEquals(createNewLine(O, O, O), rows.get(1));
+        assertEquals(createNewLine(X, O, X), rows.get(2));
     }
 
     @Test
-    public void canReturnColumns() {
-        Board board = new Board(emptyBoard);
-        List<Line> rows = board.columnLines();
-
-        assertEquals(createNewLine("1", "4", "7"), rows.get(0));
-        assertEquals(createNewLine("2", "5", "8"), rows.get(1));
-        assertEquals(createNewLine("3", "6", "9"), rows.get(2));
+    public void canReturnRowsForAFourByFour() {
+        Board board = new Board(4);
+        List<Line> rows = board.rowLines();
+        assertEquals(new Line(EMPTY, EMPTY, EMPTY, EMPTY), rows.get(0));
+        assertEquals(new Line(EMPTY, EMPTY, EMPTY, EMPTY), rows.get(1));
+        assertEquals(new Line(EMPTY, EMPTY, EMPTY, EMPTY), rows.get(2));
+        assertEquals(new Line(EMPTY, EMPTY, EMPTY, EMPTY), rows.get(3));
```

Comment:
What is this test actually testing? Your board could be returning the same row four times and this test would still pass.